src/H5Omessage.c

```diff
@@ -1740,6 +1748,27 @@ H5O__msg_alloc(H5F_t *f, H5O_t *oh, const H5O_msg_class_t *type, unsigned *mesg_
     *mesg_idx = new_idx;
 
 done:
+    /* If we are not in a recursive call, remove any deleted messages from the object header */
+    if (oh->recursion_level == 0) {
+        for (unsigned u = 0; oh->num_deleted_mesgs > 0 && u < oh->nmesgs;)
+            if (oh->mesg[u].type->id == H5O_DELETED_ID) {
+                /* Check if mesg_idx is in the section to be moved, and adjust it if so */
+                assert(u != *mesg_idx);
+                if (*mesg_idx > u)
+                    (*mesg_idx)--;
+
+                /* Slide down mesg array and adjust message counts */
+                memmove(&oh->mesg[u], &oh->mesg[u + 1], ((oh->nmesgs - 1) - u) * sizeof(H5O_mesg_t));
```

Comment:
Same comment here, should we be checking for `u < (oh->nmesgs - 1)` before indexing `&oh->mesg[u + 1]`?

Reply:
See other reply